docs/editing-components.md

```diff
@@ -0,0 +1,380 @@
+but these are the two most common locations
+where top-level Wagtail modules are stored.
+
+A simple component class looks like this:
+
+```python
+class RelatedContent(blocks.StructBlock):                      # 1
+    heading = blocks.CharBlock(required=False)                 # 2
+    paragraph = blocks.RichTextBlock(required=False)           # 3
+    links = blocks.ListBlock(atoms.Hyperlink())                # 4
+
+    class Meta:                                                # 5
+        icon = 'grip'                                          # 6
+        label = 'Related content'                              # 7
+        template = '_includes/molecules/related-content.html'  # 8
+```
+
+There are a few things happening here:
+
+1. The `RelatedContent` class is a subclass of
+   [Wagtail's `StructBlock`](https://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#structblock),
+   which allows for the combination of a fixed number of other sub-blocks
+   (see previous comment about blocks being nested within other blocks)
+   into a single unit (what we'd think of as a "module" in the Wagtail editor).
+   This one has three sub-blocks (lines 2, 3, and 4).
+2. The `heading` field uses the basic Wagtail `CharBlock`, which results in a
+   field with a basic single-line text input.
+3. The `paragraph` field uses the basic Wagtail `RichTextBlock`,
+   which results in a field with a multiline WYSIWYG text input.
+4. The `links` field uses another basic Wagtail block, `ListBlock`,
+   which is a special type of block that can hold a variable number of
+   some other block (the `Hyperlink` atom block, in this case).
+5. The `Meta` class defines some properties on the `RelatedContent` block
+   that are used by the Wagtail admin or in rendering the block.
+6. The `icon` property tells Wagtail what icon to use in the editor
+   for the button you use to add a `RelatedContent` block to a StreamField.
+   Icon options can be found in the Wagtail style guide when running locally:
+   http://localhost:8000/admin/styleguide/#icons
+7. The optional `label` property overrides the text of that same button;
+   if `label` is not set, Wagtail will generate one from the name of the block.
+8. The `template` property is a pointer to the HTML template used to render
+   this component. [See below for more on templates.](#the-html-template)
+
+This results in a module that looks like this in the Wagtail editor:
+
+![A screenshot of a Related Content module with heading, paragraph, and links fields](img/related-content-module.png)
+
+Note again that what we think of as **fields** are _also_ blocks,
+and what we think of as **components** or **modules**
+are a special kind of block, `StructBlock`,
+that comprise the sub-blocks that are our fields.
+
+There are two common optional things that are also used in component classes:
+
+1. [Overriding the default `get_context` method](http://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#streamfield-get-context)
```

Comment:
Thanks for adding this. Unfortunately this doesn't always work because of our mixed support for `{% include_block %}`. Still good to include it as a forcing factor for us to fix that.

Reply:
Yeah, I'm not gonna try to explain that in this doc :)